Eastern-Research-Group/expert-query
# Feature/164 national downloads page

## Related Issues:
* [EQ-164](https://jira.epa.gov/browse/EQ-164)

## Main Changes:
* Added a National Downloads page, which links to the files in the S3 bucket's `national-downloads` folder.
  * This page is a top-level route, so it is structured such that later projects could use the same page. The ATTAINS national downloads are listed in the **ATTAINS Data** section.
* Added a message to the download modal on the main query page that links users to the National Downloads page if queries are too large. If there are no filters on a profile query, the message will show immediately, otherwise if the count exceeds a configured number then the message will show.

## Steps To Test:
1. Create the directory `app/server/app/national-downloads`, then copy one or several of the latest gzipped national download archives (_table\_name_.csv.gz) into the folder.
2. Start the application, then navigate to http://localhost:3000/national-downloads.
3. Confirm that the archive is listed in the **ATTAINS Data** section, and take note of the file size.
4. In `app/server/app/routes/api.js`, replace line 16 with `const isLocal = false`, then reload the National Downloads page.
5. Confirm that all profile archives are listed, and that the previously noted file size matches the files size of the corresponding profile currently in the list.
6. Navigate to http://localhost:3000/attains/tmdl.
7. Without selecting any filters, click **Download**, and confirm that the download modal shows a "max size exceeded" message, with a link to the ATTAINS section on the National Downloads page.
8. Create a query that exceeds the number configured by the `maxQuerySize` variable. This is currently set as 1,000,000, so it may help to dial the number down a bit.
9. Confirm that the "max size exceeded" message shows after calculating the query's total result count.
